src/sentry_logger.c

```diff
@@ -0,0 +1,69 @@
+#include <stdio.h>
+
+#include "sentry_logger.h"
+#include "sentry_options.h"
+
+#if defined(SENTRY_PLATFORM_ANDROID)
+
+#    include <android/log.h>
+void
+sentry__logger_default(
+    sentry_level_t level, const char *message, va_list args)
+{
+    __android_log_vprint(ANDROID_LOG_INFO, "sentry-native", message, args);
+}
+
+#else
+
+void
+sentry__logger_defaultlogger(
+    sentry_level_t level, const char *message, va_list args)
+{
+    const char *prefix = "[sentry]";
+    const char *severity = sentry__logger_describe(level);
+
+    char *format = malloc(strlen(prefix) + strlen(severity) + strlen(message) + 1);
```

Comment:
We have a `sentry_malloc` which is signal safe, since we also do some logging inside signal handlers.

https://github.com/getsentry/sentry-native/blob/5020ff55ac457b02951dd56909424d0fbcda57d0/include/sentry.h#L97

Also, the prefixes need some spacing. How about a format like `[sentry] WARN …` or something like that?

Reply:
replaced with sentry, spacing added: `[sentry] priority message`